test/common/http/http1/codec_impl_test.cc

```diff
@@ -742,6 +742,17 @@ TEST_F(Http1ServerConnectionImplTest, RequestWithTrailers) {
   EXPECT_EQ(0U, buffer.length());
 }
 
+TEST_F(Http1ServerConnectionImplTest, IgnoreUpgradeH2c) {
+  initialize();
+
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer(
+      "GET http://www.somewhere.com/ HTTP/1.1\r\nConnection: "
```

Comment:
Nit: normally, in this case, the `Connection` header would have value of `Upgrade, HTTP2-Settings`. Might be worth using it here to verify that we don't accidentally remove only for `Connection: Upgrade`.

Reply:
Erm, looking at the code, I think it won't remove `Connection: Upgrade, HTTP2-Settings`.

Reply:
I think that's covered by the TODO and was already discussed between @jplevyak and @alyssawilk?

Reply:
Once more into the breach.  OK, I did the "right thing" here.  I now remove "upgrade" and "http2-settings" and only those tokens.  I also added 2 new tests for the 0 1 and 2 cases of remaining tokens.

PTAL